hedera-node/src/main/java/com/hedera/services/store/tokens/HederaTokenStore.java

```diff
@@ -333,27 +337,80 @@ public ResponseCodeEnum changeOwner(NftId nftId, AccountID from, AccountID to) {
 			final var fromThisNftsOwned = (long) tokenRelsLedger.get(fromRel, TOKEN_BALANCE);
 			final var toNftsOwned = (long) accountsLedger.get(to, NUM_NFTS_OWNED);
 			final var toThisNftsOwned = (long) tokenRelsLedger.get(asTokenRel(to, nftType), TOKEN_BALANCE);
-			nftsLedger.set(nftId, OWNER, EntityId.fromGrpcAccountId(to));
+
+			if (isKnownTreasury(to)) {
+				nftsLedger.set(nftId, OWNER, EntityId.MISSING_ENTITY_ID);
+			} else {
+				nftsLedger.set(nftId, OWNER, EntityId.fromGrpcAccountId(to));
+			}
+
 			accountsLedger.set(from, NUM_NFTS_OWNED, fromNftsOwned - 1);
 			accountsLedger.set(to, NUM_NFTS_OWNED, toNftsOwned + 1);
 			tokenRelsLedger.set(fromRel, TOKEN_BALANCE, fromThisNftsOwned - 1);
 			tokenRelsLedger.set(toRel, TOKEN_BALANCE, toThisNftsOwned + 1);
 
 			var merkleUniqueTokenId = new MerkleUniqueTokenId(fromGrpcTokenId(nftId.tokenId()), nftId.serialNo());
-			this.uniqueOwnershipAssociations.get().disassociate(
-					fromGrpcAccountId(from),
-					merkleUniqueTokenId);
+			if (!isTreasuryForToken(from, tId)) {
+				this.uniqueOwnershipAssociations.get().disassociate(
+						fromGrpcAccountId(from),
+						merkleUniqueTokenId);
+			} else {
+				this.uniqueOwnershipTreasuryAssociations.get().disassociate(
+						fromGrpcTokenId(tId),
+						merkleUniqueTokenId);
+			}
 
-			this.uniqueOwnershipAssociations.get().associate(
-					fromGrpcAccountId(to),
-					merkleUniqueTokenId);
+			if (!isTreasuryForToken(to, tId)) {
+				this.uniqueOwnershipAssociations.get().associate(
+						fromGrpcAccountId(to),
+						merkleUniqueTokenId);
+			} else {
+				this.uniqueOwnershipTreasuryAssociations.get().associate(
+						fromGrpcTokenId(tId),
+						merkleUniqueTokenId);
+			}
```

Comment:
nit: I think that it would be better if it was not `!` but rather `if(isTreasuryForToken(to, tId)`. It is easier to read imo